File: test/fixtures/Sample.java
package demo;

import demo.auth.TokenFactory;

interface Service {
    String ping();
}

class BaseService implements Service {
    public String ping() {
        return "ok";
    }
}

public class UserService extends BaseService {
    private TokenFactory factory;

    public String login(String user) {
        String token = factory.make(user);
        return store(token);
    }

    private String store(String token) {
        return token;
    }
}

class App {
    public static void main(String[] args) {
        UserService svc = new UserService();
        System.out.println(svc.login("alice"));
    }
}

File: test/fixtures/Sample.cs
using System;
using Demo.Auth;

namespace Demo
{
    interface IService
    {
        string Ping();
    }

    class BaseService : IService
    {
        public string Ping()
        {
            return "ok";
        }
    }

    public class UserService : BaseService
    {
        private TokenFactory factory;

        public string Login(string user)
        {
            var token = factory.Make(user);
            return Store(token);
        }

        private string Store(string token)
        {
            return token;
        }
    }

    class App
    {
        static void Main()
        {
            var svc = new UserService();
            Console.WriteLine(svc.Login("alice"));
        }
    }
}
